tests/cxx/tests/FVM/GeneralDiffusionTerm.cpp

```diff
@@ -25,23 +25,27 @@ GeneralDiffusionTerm::GeneralDiffusionTerm(DREAM::FVM::Grid *g, const real_t v)
 void GeneralDiffusionTerm::Rebuild(
     const real_t t, const real_t dt, DREAM::FVM::UnknownQuantityHandler*
 ) {
+    this->DiffusionTerm::ResetCoefficients();
     const len_t nr = this->grid->GetNr();
     len_t offset = 0;
 
     len_t buildIndex = static_cast<len_t>(t);
     len_t gridIndex  = static_cast<len_t>(dt);
 
-    #define SETDRR(V) if (i < np1 && j < np2) Drr(ir,i,j) = (V)
-    #define SETD11(V) if (j < np2) D11(ir,i,j) = (V)
-    #define SETD12(V) if (j < np2) D12(ir,i,j) = (V)
-    #define SETD21(V) if (i < np1) D21(ir,i,j) = (V)
-    #define SETD22(V) if (i < np1) D22(ir,i,j) = (V)
-
-    for (len_t ir = 0; ir < nr; ir++) {
-        auto *mg = this->grid->GetMomentumGrid(ir);
+    #define SETDRR(V) if (ir < nr && i < np1 && j < np2 && ir != 0) Drr(ir,i,j) = (V)
+    #define SETD11(V) if (ir < nr && i < np1 && j < np2 && i  != 0) D11(ir,i,j) = (V)
+    #define SETD12(V) if (ir < nr && i < np1 && j < np2 && i  != 0) D12(ir,i,j) = (V)
+    #define SETD21(V) if (ir < nr && i < np1 && j < np2 && j  != 0) D21(ir,i,j) = (V)
+    #define SETD22(V) if (ir < nr && i < np1 && j < np2 && j  != 0) D22(ir,i,j) = (V)
```

Comment:
Why was it necessary to extend these if statements?

Reply:
I changed it so that the flux in the terms is explicitly set to zero on the boundary, since it is strictly only then that the equation should be conservative. The alternative is to have non-zero fluxes on the boundary and then enforce a Dirichlet boundary condition where those values are ignored, but that required more finesse to work well I felt (for some reason I cannot immediately recall). If you prefer the other way, I can look into reworking it (back to the way it was originally designed).